packages/cli/generation/ir-generator/src/filterExamples.ts

```diff
@@ -0,0 +1,336 @@
+                                  ...o,
+                                  properties: o.properties
+                                      .filter((p) =>
+                                          filteredIr.hasProperty(p.originalTypeDeclaration.typeId, p.name.wireValue)
+                                      )
+                                      .map((p) => ({
+                                          ...p,
+                                          value: filterExampleTypeReference(filteredIr, p.value)
+                                      }))
+                                      .filter((p) => p.value !== undefined) as ExampleObjectProperty[]
+                              })
+                          })
+                      }),
+                      union: (u) => {
+                          const filteredUnion = filterExampleSingleUnionTypeProperties(
+                              filteredIr,
+                              u.singleUnionType.shape
+                          );
+                          return filteredUnion !== undefined
+                              ? {
+                                    ...exampleTypeReference,
+                                    shape: ExampleTypeReferenceShape.named({
+                                        ...n,
+                                        shape: ExampleTypeShape.union({
+                                            ...u,
+                                            singleUnionType: { ...u.singleUnionType, shape: filteredUnion }
+                                        })
+                                    })
+                                }
+                              : undefined;
+                      },
+                      undiscriminatedUnion: (u) => {
+                          const filteredUnion = filterExampleTypeReference(filteredIr, u.singleUnionType);
+                          return filteredUnion !== undefined
+                              ? {
+                                    ...exampleTypeReference,
+                                    shape: ExampleTypeReferenceShape.named({
+                                        ...n,
+                                        shape: ExampleTypeShape.undiscriminatedUnion({
+                                            ...u,
+                                            singleUnionType: filteredUnion
+                                        })
+                                    })
+                                }
+                              : undefined;
+                      },
+                      _other: () => {
+                          throw new Error("Received unknown type for example.");
+                      }
+                  })
+                : undefined,
+        unknown: () => exampleTypeReference,
+        _other: () => exampleTypeReference
+    });
+}
+
+function filterExamplePathParameters(
+    filteredIr: FilteredIr,
+    pathParameters: ExamplePathParameter[]
```

Comment:
you'll notice that a lot of the typescript functions in the CLI don't take parameters but take a single object with name properties instead: 

```ts
filterExamplePathParameters({ filteredIr, pathParameters }: { filteredIr: FilteredIr, pathParameters: ExamplePathParameter[]);
```

Reply:
this makes it easier to add more parameters down the road (if necessary) and also makes it more readable when you have complex expressions for the value of the parameter